### PR TITLE
CLAPPS Add create from backups tab

### PR DESCRIPTION
**Pull request by Jskobos**
## Description

Backups flow now lives at create/My Images/Backups

## Type of Change
- Non breaking change ('update', 'change')

## Note to Reviewers

Backups uses complicated logic to fetch backups for all Linodes and reshape/track the backups information. I left most of this in FromBackupsContent since it's very specific to this component. We can lift the actual network requests to the container if reviewers think it would help the experience, and possibly long-term move them to Redux. For now, this maintains the previous behavior of the app.

### CODE REVIEW

File: src/features/linodes/LinodesCreate/TabbedContent/FromBackupsContent.tsx [hunk 1]
```diff
@@ -1,545 +1,459 @@
...
+  root: {},
+  main: {},
+  sidebar: {
+    [theme.breakpoints.up('lg')]: {
+      marginTop: -130
+    }
+  }
+});
+
+export type TypeInfo =
```
Comment (martinmckenna) [inline src/features/linodes/LinodesCreate/TabbedContent/FromBackupsContent.tsx]: this looks unused

File: src/features/linodes/LinodesCreate/TabbedContent/FromBackupsContent.tsx [hunk 2]
```diff
@@ -1,545 +1,459 @@
...
+      .then(data => {
+        if (!this.mounted) {
+          return;
+        }
+        this.setState({ linodesWithBackups: data, isGettingBackups: false });
+      })
+      .catch(err => this.setState({ isGettingBackups: false }));
+  };
+
+  extendLinodes = (linodes: Linode.Linode[]): ExtendedLinode[] => {
```
Comment (martinmckenna) [inline src/features/linodes/LinodesCreate/TabbedContent/FromBackupsContent.tsx]: this function has already been abstracted to `src/features/linodes/utilities.ts`

File: src/features/linodes/LinodesCreate/TabbedContent/FromBackupsContent.tsx [hunk 3]
```diff
@@ -1,545 +1,459 @@
...
+    const label = this.label();
+
+    this.props.handleSubmitForm('createFromBackup', {
+      region: selectedRegionID,
+      type: selectedTypeID,
+      backup_id: Number(selectedBackupID),
+      label,
+      backups_enabled: backupsEnabled /* optional */,
+      booted: true,
+      tags: tagsToAdd
```
Comment (martinmckenna) [inline src/features/linodes/LinodesCreate/TabbedContent/FromBackupsContent.tsx]: missing private IP enabled

File: src/features/linodes/LinodesCreate/TabbedContent/FromBackupsContent.tsx [hunk 4]
```diff
@@ -1,545 +1,459 @@
...
+
+  componentDidMount() {
+    this.mounted = true;
+    this.getLinodesWithBackups(this.props.linodesData);
+    // If there is a selected Linode ID (from props), make sure its information
+    // is set to state as if it had been selected manually.
+  }
+
+  // Generate a default label name with a selected Linode and/or Backup name IF they are selected
+  label = () => {
```
Comment (martinmckenna) [inline src/features/linodes/LinodesCreate/TabbedContent/FromBackupsContent.tsx]: this is already being handled in the parent

File: src/features/linodes/LinodesCreate/TabbedContent/FromBackupsContent.tsx [hunk 5]
```diff
@@ -1,545 +1,459 @@
...
+      backupsEnabled
+    } = this.props;
+    const hasErrorFor = getAPIErrorsFor(errorResources, errors);
+    const generalError = hasErrorFor('none');
+
+    const imageInfo = selectedBackupInfo;
+
+    const hasBackups = backups || accountBackupsEnabled;
+
+    const label = this.label();
```
Comment (martinmckenna) [inline src/features/linodes/LinodesCreate/TabbedContent/FromBackupsContent.tsx]: label logic being handled in parent

File: src/features/linodes/LinodesCreate/TabbedContent/FromBackupsContent.tsx [hunk 6]
```diff
@@ -1,545 +1,459 @@
...
+  return linodes.filter(linode => {
+    const hasAutomaticBackups = !!linode.currentBackups.automatic.length;
+    const hasSnapshotBackup = !!linode.currentBackups.snapshot.current;
+    // backups both need to be enabled and some backups need to exist
+    // for the panel to show the Linode
+    return linode.backups.enabled && (hasAutomaticBackups || hasSnapshotBackup);
+  });
+};
+
+const formatLinodeSubheading = (typeInfo: string, imageInfo: string) => {
```
Comment (martinmckenna) [inline src/features/linodes/LinodesCreate/TabbedContent/FromBackupsContent.tsx]: abstracted in `src/features/linodes/utilities.ts`

File: src/features/linodes/LinodesCreate/TabbedContent/FromLinodeContent.tsx [hunk 7]
```diff
@@ -66,8 +66,16 @@ type CombinedProps = Props &
...
+  handleSelectLinode = (linodeID: number) => {
+    const linode = this.props.linodesData.find(
+      linode => linode.id === linodeID
+    );
+    if (linode) {
+      this.props.updateLinodeID(
+        linode.id,
+        pathOr(undefined, ['specs', 'disk'], linode)
+      );
+    }
```
Comment (martinmckenna) [inline src/features/linodes/LinodesCreate/TabbedContent/FromLinodeContent.tsx]: can't we assume the Linode exists in redux if the panel is sourcing them from Redux?

Comment (Jskobos) [reply src/features/linodes/LinodesCreate/TabbedContent/FromLinodeContent.tsx]: Probably! It crashed halfway through the PR though and I was scared to remove the safety switch.

File: src/features/linodes/LinodesCreate/TabbedContent/FromBackupsContent.tsx [hunk 8]
```diff
@@ -1,545 +1,382 @@
...
+                  typeDisplayInfo.backupsMonthly
+                ) {
+                  calculatedPrice += typeDisplayInfo.backupsMonthly;
+                }
+
+                return (
+                  <CheckoutBar
+                    heading={`${label || 'Linode'} Summary`}
+                    calculatedPrice={calculatedPrice}
+                    isMakingRequest={false}
```
Comment (martinmckenna) [inline src/features/linodes/LinodesCreate/TabbedContent/FromBackupsContent.tsx]: static

**Pull request merged**